infrastructure/lib/constructs/api-function.ts

```diff
@@ -45,13 +41,22 @@ export abstract class ApiFunction extends cdk.Construct {
    */
   public readonly route: apigw.Method;
 
+  /**
+   * The CfnFunction logicalId, used for templating the api yaml.
+   */
+  public readonly logicalId: lambda.CfnFunction;
+
   protected constructor(scope: cdk.Construct, id: string, props: ApiFunctionProps) {
     super(scope, id);
     this.method = props.method;
     this.fn = new lambdaNodeJs.NodejsFunction(this, "Function", {
       entry: props.handlerPath,
       ...props.functionPropsOverride,
     });
-    this.route = props.resource.addMethod(props.method, new apigw.LambdaIntegration(this.fn));
+    this.fn.addPermission("AllowApiGatwewayInvoke", {
+      principal: new iam.ServicePrincipal("apigateway.amazonaws.com"),
```

Comment:
Thoughts on making this a private const in the module (top level) instead?

```
const APIGW_SERVICE_PRINCIPAL = new iam.ServicePrincipal("apigateway.amazonaws.com");
```

is this better or worse?

Reply:
I don't mind that because it calls it out as a special object which may not be clear to everyone.